plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/DeployPropertyPage.java

```diff
@@ -46,17 +45,21 @@
  */
 public class DeployPropertyPage extends PropertyPage {
 
-  private DeployPreferencesPanel content;
-  private boolean canSetMessage = true;
-  private boolean isStandardPanel;
   private IFacetedProject facetedProject = null;
-  private String invalidFacetConfigErrorMessage = "";
   private static final Logger logger = Logger.getLogger(DeployPropertyPage.class.getName());
+  private FlexDeployPreferencesPanel flexPreferencesPanel;
+  private StandardDeployPreferencesPanel standardPreferencesPanel;
+  private BlankDeployPreferencesPanel blankPreferencesPanel;
```

Comment:
Could you mark somewhere with a TODO to decouple this class from the specific preference panels?  One possibility is to use an extension point where we associate a panel class with a facet type:
```
<extension point="com.google.cloud.tools.deploy.ui.properties">
  <facet id="com.google.cloud.tools.eclipse.appengine.facets.flex" version="1"
    panelClass="com.google.cloud.tools.eclipse.appengine.deploy.ui.FlexDeployPreferencesPanel" />
<extension>
```